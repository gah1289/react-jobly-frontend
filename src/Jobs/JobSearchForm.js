import React, { useEffect, useState } from 'react';
import JoblyApi from '../api';
import './Jobs.css';
import { Button } from 'reactstrap';

function JobSearchForm({ filterJobs }) {
	const INITIAL_STATE = {
		title     : undefined,
		minSalary : undefined,
		hasEquity : undefined
	};
	const [
		formData,
		setFormData
	] = useState(INITIAL_STATE);

	const [
		numResultsMsg,
		setNumResultsMsg
	] = useState();

	const handleChange = (e) => {
		let { name, value } = e.target;

		setFormData((formData) => ({ ...formData, [name]: value }));
	};

	const checkboxChange = (e) => {
		let { name, value } = e.target;
		e.target.checked ? (value = true) : (value = false);

		setFormData((formData) => ({ ...formData, [name]: value }));
	};

	const handleSubmit = async (e) => {
		e.preventDefault();
		// filter out jobs by search form data.
		const filteredJobs = await JoblyApi.getJobs({
			title     : formData.title,
			minSalary : formData.minSalary,
			hasEquity : formData.hasEquity
		});
		filterJobs(filteredJobs);
		setNumResultsMsg(`Found: ${filteredJobs.length} results`);
		setFormData(INITIAL_STATE);
	};

	return (
		<form className="jobs-form" onSubmit={handleSubmit}>
			<h2>Search by:</h2>
			<div>
				<label htmlFor="title">Title: </label>
				<input
					name="title"
					id="title"
					type="text"
					placeholder="Job Title"
					value={formData.title || ''}
					onChange={handleChange}
				/>
			</div>{' '}
			<div>
				<label>Minimum Salary: </label>
				<input
					name="minSalary"
					id="minSalary"
					type="number"
					placeholder="Minimum Salary"
					value={formData.minSalary || ''}
					onChange={handleChange}
				/>

				<label htmlFor="hasEquity">Equity </label>
				<input
					name="hasEquity"
					id="hasEquity"
					type="checkbox"
					value={formData.hasEquity || ''}
					onChange={checkboxChange}
				/>
			</div>{' '}
			<Button color="secondary">Search</Button>
			<div className="found-results">{numResultsMsg}</div>
		</form>
	);
}

export default JobSearchForm;
