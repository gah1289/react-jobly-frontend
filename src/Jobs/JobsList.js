import React, { useEffect, useState } from 'react';
import JoblyApi from '../api';
import './Jobs.css';
import { useLocation } from 'react-router-dom';
import { CardGroup } from 'reactstrap';
import { v4 as uuid } from 'uuid';

import JobCard from './JobCard';
import JobSearchForm from './JobSearchForm';

function JobsList() {
	const [
		jobs,
		setJobs
	] = useState([]);

	const [
		isLoading,
		setIsLoading
	] = useState(true);

	const filterJobs = (j) => {
		setJobs(j);
	};
	const location = useLocation();

	useEffect(() => {
		async function getJobs(data) {
			if (location.state) {
				setJobs(location.state.props);
			}
			else {
				let jobsFromApi = await JoblyApi.getJobs();
				filterJobs(jobs);
				setJobs(jobsFromApi);
			}

			setIsLoading(false);
		}
		getJobs();
	}, []);

	if (isLoading) {
		return;
	}

	return (
		<div>
			<h1>Jobs</h1>
			{!location.state ? <JobSearchForm filterJobs={filterJobs} /> : ''}
			<CardGroup>{jobs.map((job) => <JobCard key={uuid()} job={job} />)}</CardGroup>
		</div>
	);
}

export default JobsList;
