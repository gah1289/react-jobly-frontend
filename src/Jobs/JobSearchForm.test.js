import JobsList from './JobsList';
import JobSearchForm from './JobSearchForm';
import { render, screen } from '@testing-library/react';
import ItemContext from '../ItemContext';
import { MemoryRouter } from 'react-router-dom';

describe('Loads Job Card', function() {
	let isLoggedIn = true;

	let isLoading = false;

	let user = {
		username     : 'testUser',
		firstName    : 'Test',
		lastName     : 'User',
		email        : 'test@user.com',
		password     : 'password',
		isAdmin      : false,
		applications : []
	};

	user.current = user;

	let jobs = [
		{
			id            : 200,
			title         : 'Accommodation manager',
			salary        : 126000,
			equity        : null,
			companyHandle : 'mejia-scott-ryan',
			companyName   : 'Mejia, Scott and Ryan'
		},
		{
			id            : 400,
			title         : 'Accommodation manager',
			salary        : 126000,
			equity        : null,
			companyHandle : 'mejia-scott-ryan',
			companyName   : 'Mejia, Scott and Ryan'
		},
		{
			id            : 161,
			title         : 'Accountant, chartered certified',
			salary        : 86000,
			equity        : '0.070',
			companyHandle : 'boyd-evans',
			companyName   : 'Boyd-Evans'
		}
	];

	let JobsListComponent = (
		<MemoryRouter>
			<ItemContext.Provider value={{ jobs, isLoading, isLoggedIn, user }}>
				{' '}
				<JobsList>
					<JobSearchForm />
				</JobsList>{' '}
				/>{' '}
			</ItemContext.Provider>
		</MemoryRouter>
	);

	it('renders without crashing', function() {
		render(JobsListComponent);
	});

	it('matches snapshot with no jobs', function() {
		const { asFragment } = render(JobsListComponent);
		expect(asFragment()).toMatchSnapshot();
	});
});
