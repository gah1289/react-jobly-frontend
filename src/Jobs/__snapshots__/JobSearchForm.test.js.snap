// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads Job Card matches snapshot with no jobs 1`] = `
<DocumentFragment>
    /&gt; 
</DocumentFragment>
`;
